src/ApiCore/GapicClientTrait.php

```diff
@@ -400,34 +414,14 @@ function (Call $call, array $options) {
                         $startCallMethod = $this->transportCallMethods[$call->getCallType()];
                         return $this->transport->$startCallMethod($call, $options);
                     },
-                    $this->authWrapper
+                    $this->credentialsWrapper
                 ),
                 $this->agentHeaderDescriptor
             ),
             $callConstructionOptions['retrySettings']
         );
     }
 
-    /**
-     * @param array $optionalArgs {
-     *     Optional arguments
-     *
-     *     @type array $headers [optional] key-value array containing headers
-     *     @type int $timeoutMillis [optional] the timeout in milliseconds for the call
-     *     @type array $transportOptions [optional] transport-specific call options
-     * }
-     *
-     * @return array
-     */
-    protected function configureCallOptions(array $optionalArgs)
```

Comment:
What are your thoughts on removing this? It acts to restrict the options that can be passed in by the user at call time - is that something that we want? It means that any custom options will be stripped and not be processed by a custom call stack.

Reply:
Very good point - we need to either pluck them, or keep the filter.

So for an example, an argument like `withMetadata` that would be handled by a piece of custom middleware. This is implied by using the `modifyUnaryCall` method, which doesn't have access to optional args - it means that the custom call stack should always add their custom middleware/callable modifications, which should check for the argument during the call rather than when constructing the middleware.

Reply:
FYI I have just filed https://github.com/googleapis/gapic-generator/issues/1983 - we need to handle the case when optional fields clash with our supported options.

Reply:
Added a new OptionsFilterMiddleware to replace this, PTAL